backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -499,3 +516,35 @@ def handle_revoke_failure(self, error: Exception) -> None:
         DatasetAlarmService().trigger_revoke_folder_sharing_failure_alarm(
             self.target_folder, self.share, self.target_environment
         )
+
+    @staticmethod
+    def get_role_arn(target_account_id, target_requester_IAMRoleName):
+        return f"arn:aws:iam::{target_account_id}:role/{target_requester_IAMRoleName}"
+
+    @staticmethod
+    def generate_default_kms_decrypt_policy_statement(target_requester_arn):
+        return {
+            "Sid": f"{DATAALL_ACCESS_POINT_KMS_DECRYPT_SID}",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": [
+                    f"{target_requester_arn}"
```

Comment:
Same as in the BUcket sharing, what happens for the case where no KMS key is found? or there is an error in the call/permissions

Reply:
We are assuming that the following kms key is always present:
```
{
            "Sid": "Enable IAM User Permissions",
            "Effect": "Allow",
            "Principal": {
                "AWS": [
                    "arn:aws:iam::xxxxx:root",
                    "arn:aws:iam::xxxxx:role/dataallPivotRole"
                ]
            },
            "Action": "kms:*",
            "Resource": "*"
        },
```
The reason for the assumption is that this is a pre-req step to using SSE-KMS encryption keys for the share. We have to add dataallPivotRole in statement Enable IAM User Permissions while importing a dataset.

Reply:
But this might cause issues when we use S3 encryption or someone deletes the policy by mistake. I have addressed this in next revision.